src/main/java/com/netflix/blitz4j/LoggingContext.java

```diff
@@ -49,6 +56,7 @@
     private ThreadLocal<StackTraceElement> stackLocal = new ThreadLocal<StackTraceElement>();
     private ThreadLocal<LoggingEvent> loggingEvent = new ThreadLocal<LoggingEvent>();
     private ThreadLocal<Level> contextLevel = new ThreadLocal<Level>();
+    private final Set<Category> loggerNeedsLocation = Collections.synchronizedSet(Collections.newSetFromMap(new WeakHashMap<Category, Boolean>()));
```

Comment:
I'm a bit concerned about using a synchronized lock here.  Might be more performant to do something like CopyOnWriteArraySet but I'm not sure how well that would work with a WeakHashMap.  Also, why are we not caching when logger does not need location?